src/wattsi.pas

```diff
@@ -681,11 +686,61 @@    TCrossReferences = record
                            Scratch.Append($002E);
                         Scratch.Append(IntToStr(CurrentSectionNumber[CurrentHeadingRank]));
                      end;
+                     if (Variant = vDEV) then
+                     begin
+                        ID := Element.GetAttribute('id').AsString;
+                        SectionNumber := Scratch.AsString;
+                        HeadingText := DelSpace1(StringReplace(StringReplace(Element.TextContent.AsString, #$0A, ' ', [rfReplaceAll]), #$22, '\' + #$22, [rfReplaceAll]));
+                        if (not IsFirstSearchIndexItem) then
+                           Write(SearchIndexJsonFile, ',');
+                        IsFirstSearchIndexItem := False;
+                        Write(SearchIndexJsonFile, '{');
+                        Write(SearchIndexJsonFile, '"url":"' + SplitFilename + '.html#' + ID + '",');
+                        Write(SearchIndexJsonFile, '"text":"' + HeadingText + '",');
+                        if (LastDelimiter('.', SectionNumber) <> 0) then
+                        begin
+                           ParentSectionNumber := Copy(SectionNumber, 1, LastDelimiter('.', SectionNumber) - 1);
+                           ParentHeadingText := HeadingTextBySectionNumber[ParentSectionNumber];
+                           Write(SearchIndexJsonFile, '"section":"' + SectionNumber + ' ' + UTF8Encode(#$2014) + ' ' + ParentHeadingText + '"');
+                        end
+                        else
+                        begin
+                           Write(SearchIndexJsonFile, '"section":"' + SectionNumber + '"');
+                        end;
+                        Write(SearchIndexJsonFile, '}');
+                        HeadingTextBySectionNumber[SectionNumber] := HeadingText;
+                     end;
                      NewSpan := ConstructHTMLElement(eSpan);
                      NewSpan.SetAttribute('class', 'secno');
                      NewSpan.appendChild(TText.CreateDestructively(Scratch));
                      Element.InsertBefore(TText.Create(#$0020), Element.FirstChild);
                      Element.InsertBefore(NewSpan, Element.FirstChild);
+                  end
+                  else
+                  begin
+                     // TODO Find a more robust way to populate the search index
+                     // for this no-num backmatter stuff (Index, References,
+                     // Acknowledgments) because the below is brittle.
```

Comment:
Maybe explain that the reason this is necessary is because they don't have section numbers so HeadingTextBySectionNumber doesn't work